### PR TITLE
Add release action

### PR DESCRIPTION
**Pull request by thomasbtf**

### CODE REVIEW

**Review by alethomas — APPROVED** [review]

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Nice! 
Two things:

1. I think we should not have the latest case, because it kind of defeats the purpose of containerization. The goal is to not pollute container registries which images that are used by almost nobody. Only stable releases should be put there. For using the dev branch, people should rely on conda.
2. We somehow need to make sure that the containerized directive points to the latest release. This is problematic, because it has to happen before creating the release. Whenever we forget that, we fail at having the correct container annotated in the released source code. To solve this, ideally a release should be done as a PR, which can be guarded by status checks (and we could have a check for the release version being properly reflected in the containerized directive). I need to think about how that would be possible.

File: workflow/Snakefile [hunk 1]
```diff
@@ -16,6 +16,9 @@ report: "report/workflow.rst"
 container: "docker://condaforge/mambaforge"
 
 
+containerized: "quay.io/uncovar/uncovar:0.0.0"
```
Comment (thomasbtf) [inline workflow/Snakefile:19]: @johanneskoester Is this the correct format for quay.io?

**Comment by thomasbtf** [comment]
> For point 2, I think the following could work:
> 
> We add a github action that reacts on pull_request events and checks their text body. If the PR contains the words `release x.x.x`, and all other checks pass, the action
> 
> 1. modifies the containerized directive in the workflow to point to the correct docker image tag.
> 2. Automatically creates a new release. For this, the action should additionally parse the PR body for the release notes. We could have a template like this:
> 
> ```
> # Release
> x.x.x
> # Changelog
> * ...
> * ...
> ```

Nice Input! I adapted this PR to meet the criteria you listed. I hope it will work as intended. However, I don't know of a way to test this action, before pushing it to master. What would you recommend, @johanneskoester?

File: workflow/Snakefile (hunk 1 above)
Comment (johanneskoester) [reply workflow/Snakefile:19]:
```suggestion
containerized: "quay.io/uncovar/uncovar#0.0.0"
```

**Comment by johanneskoester** [comment]
> > For point 2, I think the following could work:
> > We add a github action that reacts on pull_request events and checks their text body. If the PR contains the words `release x.x.x`, and all other checks pass, the action
> > 
> > 1. modifies the containerized directive in the workflow to point to the correct docker image tag.
> > 2. Automatically creates a new release. For this, the action should additionally parse the PR body for the release notes. We could have a template like this:
> > 
> > ```
> > # Release
> > x.x.x
> > # Changelog
> > * ...
> > * ...
> > ```
> 
> Nice Input! I adapted this PR to meet the criteria you listed. I hope it will work as intended. However, I don't know of a way to test this action, before pushing it to master. What would you recommend, @johanneskoester?

I have since then found a framework for this kind of approach: https://github.com/google-github-actions/release-please-action

There are some things one needs to change with that regarding the outlined strategy. But since I am currently on vacation, let us postpone the discussion about whether we should switch and how we do that to the week of Aug 9, sorry!

**Comment by thomasbtf** [comment]
> > > For point 2, I think the following could work:
> > > We add a github action that reacts on pull_request events and checks their text body. If the PR contains the words `release x.x.x`, and all other checks pass, the action
> > > 
> > > 1. modifies the containerized directive in the workflow to point to the correct docker image tag.
> > > 2. Automatically creates a new release. For this, the action should additionally parse the PR body for the release notes. We could have a template like this:
> > > 
> > > ```
> > > # Release
> > > x.x.x
> > > # Changelog
> > > * ...
> > > * ...
> > > ```
> > 
> > 
> > Nice Input! I adapted this PR to meet the criteria you listed. I hope it will work as intended. However, I don't know of a way to test this action, before pushing it to master. What would you recommend, @johanneskoester?
> 
> I have since then found a framework for this kind of approach: https://github.com/google-github-actions/release-please-action
> 
> There are some things one needs to change with that regarding the outlined strategy. But since I am currently on vacation, let us postpone the discussion about whether we should switch and how we do that to the week of Aug 9, sorry!

No worries, have a nice vacation! 🙂

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**